feature/gribi/otg_tests/encap_frr_with_repair_vrf/README.md

```diff
@@ -0,0 +1,577 @@
+2.  Shutdown DUT port-2, port-3, and port-4.
+3.  Validate that corresponding traffic (25% of the total traffic) that was
+    encapped to 203.0.113.1 are no longer encapped, and forwarded per BGP-ISIS
+    routes (in the default VRF) out of DUT port-8.
+
+#### Test-4, primary encap unviable but backup encap viable for all tunnels
+
+Tests that if the primary NHG for all encap tunnels are unviable, then the
+traffic is re-encaped into the specified backup tunnels. This test ensures that
+the device does not withdraw this IPv4Entry and sends this traffic to routing.
+
+1.  Shutdown DUT port-2, port-3, port-4 and port-6.
+2.  Validate that traffic is encapsulated to 203.0.113.100 and 203.0.113.101 per
+    the weights.
+
+#### Test-5, primary and backup encap unviable for all tunnels
+
+Tests that if the primary NHGs of both the encap tunnel and its backup tunnel
+are unviable for all tunnels in the encap NHG, then the traffic for that cluster
+prefix is not encapped. Instead, that traffic should be forwarded according to
+the BGP/IS-IS routes in the DEFAULT VRF. This stresses the double failure
+handling, and ensures that the fallback to DEFAULT is activated through the
+backup NHGs of the tunnels instead of withdrawing the IPv4Entry.
+
+1.  Shutdown DUT port-2, port-3, port-4, port-5, port-6 and port-7.
+2.  Validate that all traffic is no longer encapsulated, and is all egressing
+    out of DUT port-8 per the BGP-ISIS routes in the default VRF.
+
+#### Test-6, primary encap unviable with backup to routing for all tunnels
+
+Tests that if the primary NHGs of both the encap tunnel is unviable, and its
+backup specifies fallback to routing, for all tunnels in the encap NHG, then the
+traffic for that cluster prefix is not encapped. Instead, that traffic should be
+forwarded according to the BGP/IS-IS routes in the DEFAULT VRF. This stresses
+the double failure handling, and ensures that the fallback to DEFAULT is
+activated through the backup NHGs of the tunnels instead of withdrawing the
+IPv4Entry.
+
+1.  Update `NHG#1000` and `NHG#1001` to the following:
+
+```
+NHG#1000 (Default VRF) { {NH#1000, DEFAULT VRF} }
+
+NH#1000 -> {
+  decapsulate_header: OPENCONFIGAFTTYPESDECAPSULATIONHEADERTYPE_IPV4
+  network_instance: "DEFAULT"
+}
+
+NHG#1001 (Default VRF) { {NH#1001, DEFAULT VRF} }
+
+NH#1001 -> {
+  decapsulate_header: OPENCONFIGAFTTYPESDECAPSULATIONHEADERTYPE_IPV4
+  network_instance: "DEFAULT"
+}
+```
+
+1.  Validate that all traffic is distributed per the hierarchical weights.
+2.  Shutdown DUT port-2, port-3, and port-4, and port-6.
+3.  Validate that all traffic is no longer encapsulated, and is all egressing
+    out of DUT port-8 per the BGP-ISIS routes in the default VRF.
+
+#### Test-7, no match in encap VRF
+
+Test that if there is no lookup match in the encap VRF, then the traffic should
+be routed to the DEFAULT VRF for further lookup.
+
+1.  In `ENCAP_TE_VRF_A`, Add an 0/0 static route pointing to the DEFAULT VRF.
```

Comment:
This is going to be done via config rightr?

Reply:
That's correct. I can add "... using gNMI" to be more prescriptive if you like.

Reply:
done